src/sources/mod.rs

```diff
@@ -15,6 +15,23 @@ pub mod ping;
 pub mod signals;
 pub mod timer;
 
+/// Possible actions that can be requested to the event loop by an
+/// event source once its events have been processed
+pub enum PostAction {
+    /// Do nothing specific
+    None,
```

Comment:
It might be confusing to name this `None`, being identical to one of the `Option` variants. Perhaps... `Continue` or `Await` or something?